### ISSUE TITLE
Some Chinese field names cannot use the cursor.execute method

### ISSUE THREAD

**Issue by wuhuanyan** [oracle/python-cx_Oracle#416]
An error occurred in executing cursor.execute.
python code:
```python
from cx_Oracle import connect, DatabaseError
from os import environ


def create_table(cursor, language):
    # set columns language
    if language == 'en':
        columns = ['A', 'B']
    elif language == 'ch1':
        columns = ['资金归属财政内设机构代码', '资金归属财政内设机构名称']
    elif language == 'ch2':
        columns = ['资金归属财政内设机构代码', '预算单位名称预算单位名称']  # Chinese field name test of the same length
    else:
        raise Exception('language not set')

    # drop table
    sql = 'drop table T_TEST_INSERT'
    try:
        cursor.execute(sql)
    except DatabaseError as e:
        if 'ORA-00942' in str(e):
            pass
        else:
            raise e

    # Create table from language
    sql = f'create table T_TEST_INSERT({columns[0]} varchar2(500 char), {columns[1]} varchar2(500 char))'
    cursor.execute(sql)


def insert(language):
    # set language
    if language == 'en':
        columns = ['A', 'B']
    elif language == 'ch1':
        columns = ['资金归属财政内设机构代码', '资金归属财政内设机构名称']
    elif language == 'ch2':
        columns = ['资金归属财政内设机构代码', '预算单位名称预算单位名称']  # Chinese field name test of the same length
    else:
        raise Exception('language not set')

    # return sql
    statement = f'INSERT INTO "T_TEST_INSERT" ("{columns[0]}", "{columns[1]}")VALUES(:"{columns[0]}", :"{columns[1]}")'
    parameters = {f'"{columns[0]}"': '1', f'"{columns[1]}"': '1'}
    return statement, parameters


def main():
    language = 'ch1'  # language = 'en' or 'ch1' or 'ch2'
    environ['NLS_LANG'] = 'SIMPLIFIED CHINESE_CHINA.UTF8'
    con = connect('find/find@ip:1521/orcl')
    cursor = con.cursor()
    create_table(cursor, language)
    statement, parameters = insert(language)
    cursor.execute(statement, parameters)
    con.commit()
    cursor.close()
    con.close()


if __name__ == '__main__':
    main()

```

In line 49, If the variable language is equal to 'ch1', Then an error will be thrown.
(Only when the variable language  is equal to 'ch1', An error occurs only when the two field names are combined.)
```python
Traceback (most recent call last):
  File "/Users/admin/Downloads/python_projects/work_project/DataTools/DataTools/tests/test_insert.py", line 62, in <module>
    main()
  File "/Users/admin/Downloads/python_projects/work_project/DataTools/DataTools/tests/test_insert.py", line 55, in main
    cursor.execute(statement, parameters)
cx_Oracle.DatabaseError: ORA-01008: 并非所有变量都已绑定
```

Here is my environmental information
```python
platform.platform: Darwin-17.4.0-x86_64-i386-64bit
sys.maxsize > 2**32: True
platform.python_version: 3.6.6
cx_Oracle.version: 6.4.1  # Latest edition 7.3.0 It's the same mistake
cx_Oracle.clientversion: (11, 2, 0, 4, 0)
Oracle Database 11g Enterprise Edition Release 11.2.0.1.0 - 64bit Production
```

**Comment by cjbj** [comment]
Thanks for the very clear testcase.  It works for me on macOS with 19.3 client and cx_Oracle 7.3:
```
platform.platform: Darwin-18.7.0-x86_64-i386-64bit
sys.maxsize > 2**32: True
platform.python_version: 3.7.7
cx_Oracle.version: 7.3.0
cx_Oracle.clientversion: (19, 3, 0, 0, 0)
```

The output from your script (with credentials changed) and a subsequent SQL*Plus query is:
```
cjones@cjones-mac2:~$ python3 issue416.py 
cjones@cjones-mac2:~$ export NLS_LANG=.AL32UTF8
cjones@cjones-mac2:~$ sqlplus -l cj/cj@localhost/orclpdb1

SQL*Plus: Release 19.0.0.0.0 - Production on Tue Apr 14 12:39:55 2020
Version 19.3.0.0.0

Copyright (c) 1982, 2019, Oracle.  All rights reserved.

Last Successful login time: Tue Apr 14 2020 12:38:01 +10:00

Connected to:
Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production
Version 19.3.0.0.0

SQL> select * from T_TEST_INSERT;

资金归属财政内设机构代码
----------------------------------------------------------------------------------------------------
资金归属财政内设机构名称
----------------------------------------------------------------------------------------------------
1
1
```

Is there any reason you are still using 11.2 client?  If your DBs are 11.2 or later, you can use 19c client.

Unzip 19c Instant Client, add a symlink to `~/lib` and see if your testcase works.  My sym link is:

```
  /Users/cjones/lib:
  total used in directory 0 available 9223372036852921570
  lrwxr-xr-x   1 cjones  staff    48 12 Nov 15:04 libclntsh.dylib -> /Users/cjones/instantclient_19_3/libclntsh.dylib
```

**Comment by wuhuanyan** [comment]
Thank you very much for your reply. I tested the 19.3 client and it can be inserted. So it has nothing to do with cx_oracle.
